### PR TITLE
SWAP-1113 Deploy 

### PR DESCRIPTION
**Pull request by peter-asztalos**
## Description

Remove unnecessary env variable and check, make token cookie readable by subdomains.

### CODE REVIEW

**Review by martin-trajanovski — APPROVED** [review]

**Comment by peter-asztalos** [comment]
@jekabs-karklins looks like the tests failing on 
```
cypress_1   |   1) Proposal administration tests
cypress_1   |        Should be able to download proposal pdf:```

with the error 

```{
  "exception": {
    "name": "JsonWebTokenError",
    "message": "jwt must be provided",
    "stack": "JsonWebTokenError: jwt must be provided
    at Object.module.exports [as verify] (/usr/src/app/node_modules/jsonwebtoken/verify.js:53:17)
    at /usr/src/app/build/src/middlewares/proposalDownload.js:328:48
    at Generator.next (<anonymous>)
    at /usr/src/app/build/src/middlewares/proposalDownload.js:27:71
    at new Promise (<anonymous>)
    at __awaiter (/usr/src/app/build/src/middlewares/proposalDownload.js:23:12)
    at /usr/src/app/build/src/middlewares/proposalDownload.js:326:62
    at Layer.handle [as handle_request] (/usr/src/app/node_modules/express/lib/router/layer.js:95:5)
    at next (/usr/src/app/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/usr/src/app/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/usr/src/app/node_modules/express/lib/router/layer.js:95:5)
    at /usr/src/app/node_modules/express/lib/router/index.js:281:22
    at param (/usr/src/app/node_modules/express/lib/router/index.js:354:14)
    at param (/usr/src/app/node_modules/express/lib/router/index.js:365:14)
    at Function.process_params (/usr/src/app/node_modules/express/lib/router/index.js:410:3)
    at next (/usr/src/app/node_modules/express/lib/router/index.js:275:10)
    at Function.handle (/usr/src/app/node_modules/express/lib/router/index.js:174:3)
    at router (/usr/src/app/node_modules/express/lib/router/index.js:47:12)
    at Layer.handle [as handle_request] (/usr/src/app/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/usr/src/app/node_modules/express/lib/router/index.js:317:13)
    at /usr/src/app/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/usr/src/app/node_modules/express/lib/router/index.js:335:12)"
  },
  "levelStr": "ERROR"
}
```

Any idea? My branch is up-to-date with development.

**Comment by martin-trajanovski** [comment]
> @jekabs-karklins looks like the tests failing on
> 
> ```
> cypress_1   |   1) Proposal administration tests
> cypress_1   |        Should be able to download proposal pdf:
> ```
> 
> with the error
> 
> ```
>   "exception": {
>     "name": "JsonWebTokenError",
>     "message": "jwt must be provided",
>     "stack": "JsonWebTokenError: jwt must be provided
>     at Object.module.exports [as verify] (/usr/src/app/node_modules/jsonwebtoken/verify.js:53:17)
>     at /usr/src/app/build/src/middlewares/proposalDownload.js:328:48
>     at Generator.next (<anonymous>)
>     at /usr/src/app/build/src/middlewares/proposalDownload.js:27:71
>     at new Promise (<anonymous>)
>     at __awaiter (/usr/src/app/build/src/middlewares/proposalDownload.js:23:12)
>     at /usr/src/app/build/src/middlewares/proposalDownload.js:326:62
>     at Layer.handle [as handle_request] (/usr/src/app/node_modules/express/lib/router/layer.js:95:5)
>     at next (/usr/src/app/node_modules/express/lib/router/route.js:137:13)
>     at Route.dispatch (/usr/src/app/node_modules/express/lib/router/route.js:112:3)
>     at Layer.handle [as handle_request] (/usr/src/app/node_modules/express/lib/router/layer.js:95:5)
>     at /usr/src/app/node_modules/express/lib/router/index.js:281:22
>     at param (/usr/src/app/node_modules/express/lib/router/index.js:354:14)
>     at param (/usr/src/app/node_modules/express/lib/router/index.js:365:14)
>     at Function.process_params (/usr/src/app/node_modules/express/lib/router/index.js:410:3)
>     at next (/usr/src/app/node_modules/express/lib/router/index.js:275:10)
>     at Function.handle (/usr/src/app/node_modules/express/lib/router/index.js:174:3)
>     at router (/usr/src/app/node_modules/express/lib/router/index.js:47:12)
>     at Layer.handle [as handle_request] (/usr/src/app/node_modules/express/lib/router/layer.js:95:5)
>     at trim_prefix (/usr/src/app/node_modules/express/lib/router/index.js:317:13)
>     at /usr/src/app/node_modules/express/lib/router/index.js:284:7
>     at Function.process_params (/usr/src/app/node_modules/express/lib/router/index.js:335:12)"
>   },
>   "levelStr": "ERROR"
> }
> ```
> 
> Any idea? My branch is up-to-date with development.

@jekabs-karklins is it something with your latest changes about pdf downloading or it might be something else?

**Comment by martin-trajanovski** [comment]
I pulled the latest develop locally into my branch and there were no failing tests. Not sure if it something with the changes that you have @peter-asztalos or it is something else.

**Comment by martin-trajanovski** [comment]
@peter-asztalos, when you visit this path locally http://localhost:3000/proposal/download/1 if you are logged in it, should download the pdf with id=1 if you are not logged in then it returns the message: {"name":"JsonWebTokenError","message":"jwt must be provided"}.

**Comment by martin-trajanovski** [comment]
Maybe you haven't updated the proposalDownload.ts middleware on the backend to use the right way of decoding the token or something like that.

**Comment by peter-asztalos** [comment]
I wonder which part caused this, will try to track down

**Comment by peter-asztalos** [comment]
Ok, I was able to track down to https://github.com/UserOfficeProject/user-office-frontend/pull/51/commits/d79eeecdab645964fa7bb8ba8b7dcc0b2226381a but looks like time to time the tests still fail and you need a re-run.

**Comment by martin-trajanovski** [comment]
> Ok, I was able to track down to [d79eeec](https://github.com/UserOfficeProject/user-office-frontend/commit/d79eeecdab645964fa7bb8ba8b7dcc0b2226381a) but looks like time to time the tests still fail and you need a re-run.

Is it just that one for proposal pdf download that is failing or something else also needs some investigation?

**Comment by peter-asztalos** [comment]
I solved the proposal PDF download, but something else failed, can't check the logs of the previous (or I don't know where they are), it was related to instruments, but after a re-run it passed, next time it happens will make a note.

File: src/components/user/SharedAuth.tsx [hunk 1]
```diff
@@ -28,13 +24,11 @@ const useStyles = makeStyles(theme => ({
...
-function checkAuthRedirect(authRedirect: string) {
+function checkAuthRedirect(authRedirect: string): boolean {
   try {
-    const url = new URL(authRedirect);
+    new URL(authRedirect);
 
-    return AUTH_ALLOWED_HOSTNAMES.some(
-      allowedHostname => allowedHostname === url.hostname
-    );
+    return true;
```
Comment (jekabs-karklins) [inline src/components/user/SharedAuth.tsx:31]: Does this mean, we are bypassing auth for now?

**Comment by peter-asztalos** [comment]
Actually using the links in the email I found the previous logs:

```
cypress_1   |   1) Instrument tests
cypress_1   |        User Officer should be able to update Instrument:
cypress_1   |      CypressError: Timed out retrying: `cy.click()` failed because this element is detached from the DOM.
cypress_1   | 
cypress_1   | `<a class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button" tabindex="0" role="button" aria-disabled="false" href="/InstrumentPage">...</a>`
cypress_1   | 
cypress_1   | Cypress requires elements be attached in the DOM to interact with them.
cypress_1   | 
cypress_1   | The previous command that ran was:
cypress_1   | 
cypress_1   |   > `cy.contains()`
cypress_1   | 
cypress_1   | This DOM element likely became detached somewhere between the previous and current command.
cypress_1   | 
cypress_1   | Common situations why this happens:
cypress_1   |   - Your JS framework re-rendered asynchronously
cypress_1   |   - Your app code reacted to an event firing and removed the element
cypress_1   | 
cypress_1   | You typically need to re-query for the element or add 'guards' which delay Cypress from running new commands.
```

After re-run it passed, we can check it in the future.

File: src/components/user/SharedAuth.tsx (hunk 1 above)
Comment (peter-asztalos) [reply src/components/user/SharedAuth.tsx:31]: No, it just doesn't really makes sense to check what is the callback URL as we don't share any information while doing the redirect, the token is set in the browser so after redirect it should be already available. The next part may be to make sure the cookie is not included when doing request to 3rd party domains.

**Review by mr-field — APPROVED** [review]

**Pull request merged**